File: purchase_requisition/purchase_requisition/custom/sales_invoice.js
frappe.ui.form.on('Sales Invoice', {
    refresh: function(frm) {
        // Set default company if not already set
        if (!frm.doc.company && frm.doc.__islocal) {
            frappe.call({
                method: 'purchase_requisition.purchase_requisition.custom.sales_invoice.get_user_default_company',
                callback: function(r) {
                    if (r.message) {
                        frm.set_value('company', r.message);
                    }
                }
            });
        }
    },
    
    company: function(frm) {
        // This function will be called when company field changes
        // You can add additional logic here if needed
        console.log('Company changed to:', frm.doc.company);
    }
}); 

File: purchase_requisition/purchase_requisition/custom/sales_invoice.py
import frappe

@frappe.whitelist()
def get_user_default_company():
    """Get the default company for the current user"""
    user_company = frappe.get_value("User", frappe.session.user, "company")
    return user_company

@frappe.whitelist()
def set_default_company(doc, method):
    """Hook function to set default company"""
    if not doc.company:
        user_company = frappe.get_value("User", frappe.session.user, "company")
        if user_company:
            doc.company = user_company

def before_insert(doc, method):
    """Set default company before insert"""
    if not doc.company:
        user_company = frappe.get_value("User", frappe.session.user, "company")
        if user_company:
            doc.company = user_company

def onload(doc, method):
    """Set default company on form load"""
    if not doc.company and doc.is_new():
        user_company = frappe.get_value("User", frappe.session.user, "company")
        if user_company:
            doc.company = user_company 